enterprise/mqtt/src/main/java/io/crate/mqtt/operations/MqttIngestService.java

```diff
@@ -86,26 +141,58 @@ private static String mqttPayloadToString(ByteBuf content) {
     }
 
     public void doInsert(String clientId, MqttPublishMessage msg, BiConsumer<Object, Throwable> ackCallback) {
+        if (isInitialized == false) {
+            throw new IllegalStateException("Service was not initialized");
+        }
+
         Map<String, Object> payload = parsePayloadToMap(msg.content());
         if (payload == null) {
             return;
         }
-        List<Object> args = Arrays.asList(clientId,
+
+        Set<Tuple<Predicate<Row>, IngestRule>> predicateAndIngestRules = predicateAndIngestRulesReference.get();
+        Object[] args = new Object[]{clientId,
             msg.variableHeader().packetId(),
             msg.variableHeader().topicName(),
-            payload
-        );
+            payload};
+        List<Object> argsAsList = Arrays.asList(args);
+
+        for (Tuple<Predicate<Row>, IngestRule> entry : predicateAndIngestRules) {
+            if (entry.v1().test(new RowN(args))) {
+                IngestRule ingestRule = entry.v2();
+                SQLOperations.Session session = createSessionFor(TableIdent.fromIndexName(ingestRule.getTargetTable()));
+                session.bind(SQLOperations.Session.UNNAMED, STATEMENT_NAME, argsAsList, null);
+                ResultReceiver resultReceiver = new BaseResultReceiver();
+                resultReceiver.completionFuture().whenComplete(ackCallback);
+                session.execute(SQLOperations.Session.UNNAMED, 0, resultReceiver);
+                session.sync();
+                session.close();
+            }
+        }
         if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Parsed MQTT message into arguments: {}", args);
+            LOGGER.debug("Parsed MQTT message into arguments: {}", argsAsList);
         }
-        session.bind(Session.UNNAMED, STMT_NAME, args, null);
         ResultReceiver resultReceiver = new BaseResultReceiver();
         resultReceiver.completionFuture().whenComplete(ackCallback);
-        session.execute(Session.UNNAMED, 0, resultReceiver);
-        session.sync();
     }
 
-    public void close() {
-        session.close();
+    @Override
+    public void applyRules(Set<IngestRule> rules) {
```

Comment:
Maybe rename this to `onRulesUpdate` or something like that? To make it clearer that this can be called multiple times

Reply:
We started with something similar but felt odd to have it named as `onRulesChanged`\`onRulesUpdate`\`onRules` as the client receives *all* rules where there is a change regarding any rule. 
`applyRules` seemed to be a good compromise to express this and move away from the observer like namings where only the changes are published to the interested parties.